File: src/main/java/recipeapplication/model/RecentlyViewed.java
package recipeapplication.model;

import javax.persistence.*;

@Entity
@Table(name = "recentlyViewed")
public class RecentlyViewed
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "recipe")
    private Recipe recipe;

    @OneToOne
    private User user;

    public User getUser()
    {
        return user;
    }

    public void setUser(User user)
    {
        this.user = user;
    }

    public Recipe getRecipe()
    {
        return recipe;
    }

    public void setRecipe(Recipe recipe)
    {
        this.recipe = recipe;
    }
}

File: src/main/kotlin/recipeapplication/model/RecentlyViewed.kt
package recipeapplication.model

import javax.persistence.*

@Entity
@Table(name = "recentlyViewed")
class RecentlyViewed {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Long? = null

    @ManyToOne
    @JoinColumn(name = "recipe")
    lateinit var recipe: Recipe

    @OneToOne
    lateinit var user: User
}